diff --git a/browser.js b/browser.js
index 0fb0b7153f8ae86217ceca06b2385795ca669877..9a8f4b193a54fd96145e7bafd3e98d62f876c560 100644
--- a/browser.js
+++ b/browser.js
@@ -2,18 +2,18 @@
 
 // limit of Crypto.getRandomValues()
 // https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues
-var MAX_BYTES = 65536
+const MAX_BYTES = 65536
 
 // Node supports requesting up to this number of bytes
 // https://github.com/nodejs/node/blob/master/lib/internal/crypto/random.js#L48
-var MAX_UINT32 = 4294967295
+const MAX_UINT32 = 4294967295
 
 function oldBrowser () {
   throw new Error('Secure random number generation is not supported by this browser.\nUse Chrome, Firefox or Internet Explorer 11')
 }
 
-var Buffer = require('safe-buffer').Buffer
-var crypto = global.crypto || global.msCrypto
+const _global = typeof globalThis !== 'undefined' ? globalThis : global
+const crypto = _global.crypto || _global.msCrypto
 
 if (crypto && crypto.getRandomValues) {
   module.exports = randomBytes
@@ -25,12 +25,12 @@ function randomBytes (size, cb) {
   // phantomjs needs to throw
   if (size > MAX_UINT32) throw new RangeError('requested too many random bytes')
 
-  var bytes = Buffer.allocUnsafe(size)
+  const bytes = Buffer.allocUnsafe(size)
 
-  if (size > 0) {  // getRandomValues fails on IE if size == 0
+  if (size > 0) { // getRandomValues fails on IE if size == 0
     if (size > MAX_BYTES) { // this is the max bytes crypto.getRandomValues
       // can do at once see https://developer.mozilla.org/en-US/docs/Web/API/window.crypto.getRandomValues
-      for (var generated = 0; generated < size; generated += MAX_BYTES) {
+      for (let generated = 0; generated < size; generated += MAX_BYTES) {
         // buffer.slice automatically checks if the end is past the end of
         // the buffer so we don't have to here
         crypto.getRandomValues(bytes.slice(generated, generated + MAX_BYTES))
